### PR TITLE
Piper/remove evm stuff from peer and peer pool

### PR DESCRIPTION
**Pull request by pipermerriam**
### What was wrong?

WIP: Alternate approach to https://github.com/ethereum/py-evm/pull/1258

### How was it fixed?

Still a WIP but:

- removes all of the EVM/Trinity stuff from `p2p.peer.BasePeer` and `p2p.peer.PeerPool`
- creates new base `trinity.protocol.common.peer.TrinityPeer` which both `ETHPeer` and `LESPeer` inherit from.

When starting a peer, the peer pool now waits for `peer.booted: asyncio.Event` prior to peer being added to the peer pool.

Peer classes may now implement an `async def _boot(self):` method which is run asynchronously at the beginning of `BasePeer._run` using the `run_task` API.  The default implementation of `_boot` is a noop.

The new `TrinityPeer` class implements the dao fork check within it's own `_boot`.

### Other cleanup

All of the following should/will-probably be extracted from this PR

- removal of sharding code
- removal of `LESPeer.head_info` object

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by carver** [comment]
I can't really read the diff with all the sharding and other refactoring going in. But first thought is that `TrinityPeer` doesn't feel like the right name. Would a swarm peer be a subclass of `TrinityPeer`? Maybe it's an `EVMPeer` that is the superclass of eth and les peers.

File: p2p/peer.py [hunk 1]
```diff
@@ -978,140 +921,3 @@ def __init__(self, peers: Tuple[BasePeer, ...]) -> None:
...
-                 block_hash: Hash32,
-                 total_difficulty: int,
-                 genesis_hash: Hash32) -> None:
-        self.block_number = block_number
-        self.block_hash = block_hash
-        self.total_difficulty = total_difficulty
-        self.genesis_hash = genesis_hash
-
-
-def _test() -> None:
```
[the remark this answers is not in the document]
Comment (gsalgado) [reply p2p/peer.py]: It actually is; I use it quite often to connect directly to geth (either my local instance or a remote one) and test/debug issues

**Comment by pipermerriam** [comment]
@carver yeah, looking at it now it is still pretty hard to read even with the sharding stuff gone.  I'll work on cleaning up more today.

File: p2p/peer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pipermerriam) [reply p2p/peer.py]: np, I'll restore it.

**Comment by pipermerriam** [comment]
closing for now in favor of #1265 

I'll be re-working this branch after that is merged and will open a new PR.

**Pull request closed without merging**